rebasehelper/checkers/abipkgdiff_tool.py

```diff
@@ -113,19 +116,90 @@ def run_check(cls, result_dir, **kwargs):
-            reports[output] = text
-        reports['critical_changes'] = cls.critical_changes
-        return reports
+            ret_code = int(ret_code)
+            reports[old_name] = ret_code
+        return {'packages': cls.parse_abi_logs(reports), 'abi_changes': cls.abi_changes, 'path': cls.results_dir}
+
+    @classmethod
+    def parse_abi_logs(cls, reports):
+        """
+        Parse abipkgdiff logs
+        :param reports: dictionary with paths to the logs of the produced (sub)packages
+        :return: returns packages names with its abipkgdiff changes
+        """
+        pkgs = {}
+        for pkg, ret_code in six.iteritems(reports):
+            cur_pkg = {}
+            if ret_code:
+                with open(os.path.join(cls.results_dir, pkg + '.txt'), 'r') as f:
+                    lines = f.readlines()
+                    cur_pkg['functions_changes'] = cls.parse_line("Functions changes .* (\d+) .* (\d+) .* (\d+)",
+                                                                 lines[1])
+                    cur_pkg['variables_changes'] = cls.parse_line("Variables changes .* (\d+) .* (\d+) .* (\d+)",
+                                                                 lines[2])
+                    cur_pkg['functions_symbols_changes'] = cls.parse_line("Function symbols .* (\d+) .* (\d+)",
+                                                                         lines[3])
+                    cur_pkg['variables_symbols_changes'] = cls.parse_line("Variable symbols .* (\d+) .* (\d+)",
+                                                                          lines[4])
+                    cls.abi_changes = True
+            # If no abi changes for the package, store empty dictionary
+            pkgs[pkg] = cur_pkg
+        return pkgs
+
+    @classmethod
+    def parse_line(cls, regex, line):
+        """
+        Parse abipkgdiff line
+        :param regex: Regex used for info gathering
+        :param line: line containing the needed information
+        :return: list with found abipkgdiff changes
+        """
+        found = re.search(regex, line)
+        if len(found.groups()) == 2:
+            if 'function' in line:
+                return ['%s function symbol(s) removed' % found.group(1),
+                        '%s function symbol(s) added' % found.group(2)]
+            elif 'variable' in line:
+                return ['%s variable symbol(s) removed' % found.group(1),
+                        '%s variable symbol(s) added' % found.group(2)]
+        elif len(found.groups()) == 3:
+            if 'function' in line:
+                return ['%s function(s) removed' % found.group(1), '%s function(s) changed' % found.group(2),
+                        '%s function(s) added' % found.group(3)]
+            elif 'variable' in line:
+                return ['%s variable(s) removed' % found.group(1), '%s variable(s) changed' % found.group(2),
+                        '%s variable(s) added' % found.group(3)]
```

Comment:
I don't like this method. You already know which of the regexps matched, so why test count of groups and presence of some string?
Also, what if regexp doesn't match at all?